src/reconstruction/comparison/modelBorgifier/src/compareAllMets.m

```diff
@@ -0,0 +1,194 @@
+% This file is published under Creative Commons BY-NC-SA.
+%
+% Please cite:
+% Sauls, J. T., & Buescher, J. M. (2014). Assimilating genome-scale 
+% metabolic reconstructions with modelBorgifier. Bioinformatics 
+% (Oxford, England), 30(7), 1036?8. http://doi.org/10.1093/bioinformatics/btt747
+%
+% Correspondance:
+% johntsauls@gmail.com
+%
+% Developed at:
+% BRAIN Aktiengesellschaft
+% Microbial Production Technologies Unit
+% Quantitative Biology and Sequencing Platform
+% Darmstaeter Str. 34-36
+% 64673 Zwingenberg, Germany
+% www.brain-biotech.de
+%
+function metScores = compareAllMets(Cmodel, Tmodel)
+% compareAllMets compares and scores similarity between all pairwise
+% combinations of metabolites between two models. It is not however used
+% by the main workflow. 
+%
+% USAGE:
+%    metScores = compareAllMets(Cmodel,Tmodel
+%
+% INPUTS:
+%    Cmodel:    Comparison model. 
+%    Tmodel:    Template model. 
+%
+% OUTPUTS:
+%    metScore:  Matrix of scores of all pairwise comparisons. 
+%
+% CALLS:
+%    None
+%
+% CALLED BY:
+%    None
+
+
+%% Declare variables and scoring structure.
+nCmets = length(Cmodel.mets) ;
+nTmets = length(Tmodel.mets) ;
+
+% Score values.
+ScoreVal = struct('ID', [10,-1], ...
+                  'compartment', [0,-100], ...
+                  'name', [10,-1], ...
+                  'formula', [10,-5], ...
+                  'charge', [1,-1], ...
+                  'KEGGID', [10,-5], ...
+                  'SEEDID', [10,-5], ...
+                  'ChEBIID', [10,-5], ...
+                  'PubChemID', [10,-5], ...
+                  'InChIString', [10,-5]) ;
+
```

Comment:
General comment here (even though it should not apply, since ChEBIIDs are actually nto checked for all I see:
Ids can have multiple entries separated by ;. So it would be necessary to check whether they have ANY ID in common.